code_annotations/extensions/plaintext.py

```diff
@@ -0,0 +1,76 @@
+"""
+Stevedore extension for static annotation searching in any plain text file
+"""
+
+import re
+
+import six
+
+from .base import AnnotationExtension
+
+
+class PlaintextAnnotationExtension(AnnotationExtension):
+    """
+    Extension for any type of text files
+    """
+    extension_name = 'plaintext'
+
+    def __init__(self, config):
+        super(PlaintextAnnotationExtension, self).__init__(config)
+
+        strings_to_search = []
+        annotation_tokens = config['annotations']
+
+        # Take the configured annotations and turn them into a regex for our search
+
+        # TODO: Refactor this wall of text and extend to enforce groups
+        for annotation_or_group in annotation_tokens:
+            if isinstance(annotation_tokens[annotation_or_group], six.string_types):
+                strings_to_search.append(annotation_tokens[annotation_or_group])
+            elif isinstance(annotation_tokens[annotation_or_group], (list, tuple)):
+                group = annotation_tokens[annotation_or_group]
+                for annotation in group:
+                    if isinstance(annotation, six.string_types):
+                        strings_to_search.append(annotation)
+                    elif isinstance(annotation, dict):
+                        strings_to_search.append(list(annotation.keys())[0])
+                    else:
+                        raise TypeError('{} is an unknown type. Annotations must be strings or list/tuples.'.format(
+                            annotation_tokens[annotation_or_group])
+                        )
+            else:
+                raise TypeError('{} is an unknown type. Annotations must be strings or list/tuples.'.format(
+                    annotation_tokens[annotation_or_group])
+                )
+
+        self.query = r'{}'.format('|'.join(strings_to_search))
+
+        if config['verbose'] > 1:
+            print("Plaintext extension regex query: {}".format(self.query))
+
+    def validate(self, file_handle):
+        # TODO: Implement!
+        return True
+
+    def search(self, file_handle):
+        txt = file_handle.read()
+        found_annotations = []
+
+        # Fast out if no annotations exist in the file
+        if re.search(self.query, txt):
+            line_num = 0
+            for line in txt.splitlines():
+                line_num += 1
+                for m in re.finditer(self.query, line):
+                    token = line[m.start():m.end()].strip()
+                    substring = line[m.end():None].strip()
```

Comment:
I have a hunch we're just going to re-use this chunk of code for parsing model docstrings, so maybe this would be best defined in a shared function in another file?  Like, a function which takes `query` and `line_iterator` as args?

Reply:
I'm not sold that this is the way to go for the implementation yet, but if it ends up being useful in another place that would be the time to move it to a shared location. I just added a `helpers.py` that might work.